vnext/ReactUWP/Views/ReactControl.cpp

```diff
@@ -235,5 +258,40 @@ int64_t ReactControl::GetActualWidth() const {
   return static_cast<int64_t>(element.ActualWidth());
 }
 
+void ReactControl::PrepareXamlRootView(XamlView const& rootView) {
+  if (auto panel = rootView.try_as<winrt::Panel>()) {
```

Comment:
>if (auto panel = rootView.try_as<winrt::Panel>()) { [](start = 2, length = 51)

IIUC, if the app that's hosting ReactControl doesn't put create a Panel to hold the ReactControl, we won't be able to create a safe harbor.  Is that correct?  If so, this seems like an unnecessary limitation.  Why not make the safe harbor the child of m_xamlRootView, along with its other special children?  Then we would have no limitation.

Reply:
Technically we can support both Grid/Panel and ContentControl as the parent of ReactControl.
Only Panel has Children property, since we don't have parent which is ContentControl yet, I didn't add that part of code. 

The prototype version is like what you said, rootview->contentcontrol->children's. it cause some layout problem because different combination of alignment/contentalignment setting for rootview and contentcontrol.

I will try current implementation.